src/oled_display_node.cpp

```diff
@@ -775,148 +773,162 @@ void displayApiCallback(const oled_display_node::DisplayOutput::ConstPtr& msg)
+    // Battery_state topic and we then get callbacks for each message
+    ros::Subscriber sub2 = nh.subscribe("battery_state", 100, batteryStateApiCallback);
 
-        // If you change the text, use same number of chars so display does not move around on the line
-        std::stringstream powStream;
-	if (g_motorPowerActive > 0) {
-            powStream <<   " ON";
-        }else {
-            powStream <<   "OFF";
+    // Motor_power_active topic
+    ros::Subscriber sub3 = nh.subscribe("motor_power_active", 100, motorPowerActiveApiCallback);
+
+    // Firmware state topic
+    ros::Subscriber sub4 = nh.subscribe("firmware_version", 100, firmwareVersionCallback);
+
+    // We will refresh the lines from time to time in case IP addr has changed
+    ros::Rate loop_rate(0.5);
+    int32_t loopIdx = 0;
+
+    // mainloop:
+    while ((dispError == 0) && ros::ok())
+    {
+        loopIdx++;
+        hostname = getPopen("uname -n");
+        dispError |= dispOled_writeText(&g_oledDisplayCtx, DISP_LINE_HOSTNAME, 0, DISP_TEXT_START_MODE, hostname.c_str());
+        ros::Duration(updateDelay).sleep();
+
+        getIpAddressses(displayedETHAddress, displayedWLANAddress, 0);
+        dispError |= dispOled_writeText(&g_oledDisplayCtx, DISP_LINE_IP_WLAN, 0, DISP_TEXT_START_MODE, displayedWLANAddress.c_str());
+        ros::Duration(updateDelay).sleep();
+        dispError |= dispOled_writeText(&g_oledDisplayCtx, DISP_LINE_IP_ETH, 0, DISP_TEXT_START_MODE, displayedETHAddress.c_str());
+        ros::Duration(updateDelay).sleep();
+
+        // If there is a battery_state topic and we get the callback also show battery voltage
+        if (g_batteryVoltage > 0.0) {
+            ROS_DEBUG("%s Battery voltage is now %4.1f volts.", THIS_NODE_NAME, g_batteryVoltage);
+
+            // Control what shows up after the voltage reading.
+            // If you change the text, use same number of chars so display does not move around on the line
+            std::stringstream stream;
+            stream << std::fixed << std::setprecision(1) << g_batteryVoltage;
+
+            if (g_batteryVoltage <= BAT_LOW_LEVEL && (loopIdx % 2) == 0){
+                stream << "V  LOW";
+            }
+            else if (g_batteryVoltage >= BAT_CHARGING_LEVEL && (loopIdx % 3) != 0){
+                stream << "V CHRG";
+            }
+            else{
+                stream << "V" << g_batteryPercentage;
+            }
+
+            std::string battText = "Bat:" + stream.str();
+            dispError |= dispOled_writeText(&g_oledDisplayCtx, DISP_LINE_BATT_VOLTS, 1, DISP_TEXT_START_MODE, battText.c_str());
+            ros::Duration(updateDelay).sleep();
         }
-        std::string motPowerText = "Mot Power " + powStream.str();
+
+        // If you change the text, use same number of chars so display does not move around on the line
+        std::stringstream infoStream;
+        infoStream << "Mot:" << g_motorPowerActive << " FW:" << g_firmwareVersion;
```

Comment:
Ok.  I guess --- is ok for indication of no callback.   This will in fact happen to people who enable the display and pull new code so this will not just be for leaches who the oled driver.   I silenced some parts of the display s that blank was better than stuff but the argument should be we do what is forward looking for the product so it's ok.

Reply:
Yeah we could easily do blank I suppose, but this seemed more indicative at the time.